src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -431,27 +423,129 @@ CreateShardPlacements(Oid relationId, int64 shardId, List *ddlEventList,
 
 
 /*
+ * InsertShardPlacementRows inserts shard placements to the metadata table on
+ * the coordinator node. Then, returns the list of added shard placements.
+ */
+List *
+InsertShardPlacementRows(Oid relationId, int64 shardId, List *workerNodeList,
```

Comment:
I think functions of master_create_shards.c consists of higher level callers, so remaining that function in master_stage_protocol.c is better idea.